From 0b992c34b95735083e8c3d6b815fdd43ca1a2eaa Mon Sep 17 00:00:00 2001
Message-Id: <0b992c34b95735083e8c3d6b815fdd43ca1a2eaa.1417465742.git.chang-joon.lee@intel.com>
In-Reply-To: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
References: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Wed, 19 Nov 2014 11:48:38 +0530
Subject: [PATCH 11/11] REVERTME [VPG]: drm/i915/vlv: Support fb width and
 height less than 3 in plane updates.

Add support in plane update to accept fb width and height < 3 for VLV
based devices.
User layer can send fb width and height < 3 which are valid
for plane updates in some instances and we do not want to
discard the update.

Patch is currently REVERTME as it is being reviewed if the limits can be relaxed for all platforms.

Issue: GMINL-4142
Change-Id: I5c9d8d30730cbb64484d85722662a55ff3bad298
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_sprite.c |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index d131da9..650be2a 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -1444,8 +1444,18 @@ intel_update_plane(struct drm_plane *plane, struct drm_crtc *crtc,
 
 	/* FIXME check all gen limits */
 	if (fb->width < 3 || fb->height < 3 || fb->pitches[0] > 16384) {
-		DRM_DEBUG_KMS("Unsuitable framebuffer for plane\n");
-		return -EINVAL;
+		/*
+		 * User layer can send width/height < 3 in few instances
+		 * Relaxing these limits for all platforms are being
+		 * considered. But for now, do it only for VLV
+		 * based devices.
+		 */
+		if (IS_VALLEYVIEW(dev) && fb->pitches[0] <= 16384)
+			DRM_DEBUG_KMS("Allow lesser fb width/height\n");
+		else {
+			DRM_DEBUG_KMS("Unsuitable framebuffer for plane\n");
+			return -EINVAL;
+		}
 	}
 
 	/* Sprite planes can be linear or x-tiled surfaces */
-- 
1.7.9.5

